testsuite/classlibrary/TestServer_clientID_booted.sc

```diff
@@ -70,4 +70,128 @@ TestServer_clientID_booted : UnitTest {
 		s.quit;
 		s.remove;
 	}
+
+	// test that repeated login from the same server object to a running server process
+	// ends with correct server status, and keeps clientID and allocators  as is.
+	test_repeatedLogin {
+		var server, cond, timer, lastNodeID;
+
+		server = Server(thisMethod.name.asSymbol);
+		server.options.maxLogins = 4;
+		server.clientID = 3;
+		this.bootServer(server);
+
+		// use some nodeIDs so we know whether nodeID allocator is reset
+		10.do { server.nextNodeID };
+		lastNodeID = server.nextNodeID;
+
+		// simulate losing network contact by turning off watcher,
+		// but actually leave the server process running
+		server.statusWatcher
+			.stopStatusWatcher
+			.stopAliveThread
+			.notified_(false)
+			.serverRunning_(false);
+
+//		1.wait;
+
+		// now login again from the same server object
+		// should get "This seems to be a login after loss of network ... all is well" post.
+		server.startAliveThread;
+
+		cond = Condition.new;
+
+		timer = fork { 3.wait; cond.unhang };
+		server.doWhenBooted {
+			"% - server login timed out.\n".postf(thisMethod);
+			timer.stop;
+			cond.unhang;
+		};
+
+		cond.hang;
+
+		this.assert(server.serverRunning,
+			"after repeated login, server should be known to be fully running."
+		);
+
+		this.assert(server.clientID == 3,
+			"after repeated login, remote client should get the same clientID from server process."
+		);
+
+		this.assert(server.nextNodeID > lastNodeID,
+			"after repeated login, nodeID allocator should not be reset."
+		);
+
+		server.quit;
+		server.remove;
+
+	}
+
+	// test that logins to a (pseudo-) remote server process work as intended
+	// first login initializes fully,
+	// second login from same address gets the same clientID, and sets running state.
+	test_remoteLogins {
+		var options, serverPid, remote1, remote2, cond, timer;
+
+		options = ServerOptions.new;
+		options.maxLogins = 4;
+
+		// simulate a remote server process by starting a server process independently of SC
+		serverPid = unixCmd(Server.program + options.asOptionsString);
+
+		1.wait;
```

Comment:
what is this wait for?

Reply:
leftover from an earlier version of the test, unneeded now - can remove it.